terraform/modules/marklogic/marklogic_iam.tf

```diff
@@ -32,27 +32,36 @@ resource "aws_iam_policy" "ml_instance_policy" {
   path        = "/"
   description = "Allows MarkLogic instances to perform necessary actions"
 
-  # Terraform's "jsonencode" function converts a
-  # Terraform expression result to valid JSON syntax.
   policy = jsonencode({
     Version = "2012-10-17"
     Statement = [
       {
         Action = [
           "ec2:DescribeInstances",
+          "ec2:DescribeVolumes",
           "ec2messages:GetMessages",
+          "ec2:CreateTags",
 
           "ssm:UpdateInstanceInformation",
           "ssm:ListInstanceAssociations",
           "ssm:ListAssociations",
           "ssm:PutInventory",
           "ssm:UpdateInstanceAssociationStatus",
           "ssmmessages:OpenControlChannel",
-          "ssmmessages:CreateControlChannel"
+          "ssmmessages:CreateControlChannel",
         ]
         Effect   = "Allow"
         Resource = "*"
       },
+      {
+        Action = [
+          "kms:GenerateDataKey",
+          "kms:DescribeKey",
+          "kms:Decrypt"
+        ]
+        Effect   = "Allow"
+        Resource = [aws_kms_key.ml_logs_encryption_key.arn]
+      },
```

Comment:
@hugh-emerson I've restricted this to just the logs key as I think that's all it needs, is that right?